### PR TITLE
Cosine extension

### PR DESCRIPTION
**Pull request by florian-huber**
- fix wrong denominator in cosine score calculation
- add options to raise peaks to given power

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,13 +9,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Added peak weighting option for cosine score and modified cosine score [#96](https://github.com/matchms/matchms/issues/96)
 - Added similarity score based on comparing parent masses [#79](https://github.com/matchms/matchms/pull/79)
```
Comment (sverhoeven) [inline CHANGELOG.md]: Dont need so many added
```suggestion
- Peak weighting option for cosine score and modified cosine score [#96](https://github.com/matchms/matchms/issues/96)
- Similarity score based on comparing parent masses [#79](https://github.com/matchms/matchms/pull/79)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,13 +9,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Changed
 
 - CosineGreedy function is now numba based [#86](https://github.com/matchms/matchms/pull/86)
 - Extended readthedocs documentation [#82](https://github.com/matchms/matchms/issues/82)
 
+### Fixed
+
+- Fixed incorrect denominator for cosine score normalization [#98](https://github.com/matchms/matchms/pull/98)
```
Comment (sverhoeven) [inline CHANGELOG.md]: Same as before
```suggestion
- Incorrect denominator for cosine score normalization [#98](https://github.com/matchms/matchms/pull/98)
```

File: matchms/similarity/CosineGreedy.py [hunk 3]
```diff
@@ -42,17 +42,24 @@ class CosineGreedy:
...
+    def __init__(self, tolerance: float = 0.1, mz_power: float = 0.0,
+                 intensity_power: float = 1.0):
         """
         Parameters
         ----------
-        tolerance
-            Peaks will be considered a match when <= tolerance apart.
+        tolerance:
+            Peaks will be considered a match when <= tolerance apart. Default is 0.1.
+        mz_power:
```
Comment (sverhoeven) [inline matchms/similarity/CosineGreedy.py:55]: x**0 always gives 1. 

Is it correct if `mz_power=0 ` then mz is ignored during score generation? 
If yes, is that clear to library users or should this be documented?

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Looks good to me, adding new arguments with defaults should not break anything.

**Comment by florian-huber** [comment]
We suddenly get another ``pylint`` error. Seems related to a bug https://github.com/PyCQA/pylint/issues/2767

File: matchms/similarity/CosineGreedy.py (hunk 3 above)
Comment (florian-huber) [reply matchms/similarity/CosineGreedy.py:55]: Thanks Stefan. I added such an explanation to ``CosineGreedy`` and ``ModifiedCosine``.

File: matchms/similarity/spectrum_similarity_functions.py [hunk 4]
```diff
@@ -18,6 +20,10 @@ def collect_peak_pairs(spec1, spec2, tolerance, shift=0):
         Peaks will be considered a match when <= tolerance appart.
     shift : float, optional
         Shift spectra peaks by shift. The default is 0.
+    mz_power: float, optional
+        The power to raise mz to in the cosine function. The default is 0.
```
Comment (sverhoeven) [inline matchms/similarity/spectrum_similarity_functions.py]: For consistency
```suggestion
        The power to raise mz to in the cosine function. The default is 0, in which
        case the peak intensity products will not depend on the m/z ratios.
```

**Review by sverhoeven — APPROVED** [review]
OK to merge for me , just one suggestion for consistency.

**Pull request merged**